File: leetcode/49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {

        unordered_map<string,vector<string>> mp;
        vector<vector<string>> v;
        for(string s : strs){
             string sorteds = s;
             sort(sorteds.begin(),sorteds.end());

             mp[sorteds].push_back(s);
        }

        for(auto it : mp){
              vector<string> c;
             for(auto str : it.second){
                  c.push_back(str);
             }
             v.push_back(c);
        }

        return v;
        
    }
};

File: leetcode/49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        
        Map<String,List<String>> mp = new HashMap<>();
        for(String str : strs){
            String keyStr = str;
            char[] keyArr = str.toCharArray();
            Arrays.sort(keyArr);
            String sortedKeyStr = new String(keyArr);

            mp.computeIfAbsent(sortedKeyStr, k -> new ArrayList<>()).add(str);
        }
        List<List<String>> ls = new ArrayList<>(mp.values());
        return ls;
        
    }
}